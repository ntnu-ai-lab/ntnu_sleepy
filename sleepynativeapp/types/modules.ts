import { User } from "./Types";

export interface Answer {
  id?: string;
  input: string;
  answerList?: string;
  value: string | number | boolean;
}
export interface AnswerList {
  id?: string;
  section: string;
  user: string;
  answers: Answer[];
}
export interface Input {
  id: string;
  type: string;
  name: string;
  label: string;
  helptext: string;
  value: string;
  section: string;
  options?: FormSelectOption[];
  answers?: Answer[];
}

export interface FormSelectOption {
  value: string;
  label: string;
}

export interface Question {
  options: QuizOption[];
  question: string;
}

export interface QuizOption {
  label: string;
  correct: boolean;
}
export interface Module {
  title: string;
  id: string;
  parts: string[];
}

export interface ModuleExpanded {
  title: string;
  id: string;
  parts: Part[];
}

export interface Part {
  title: string;
  id: string;
  module: string;
  pages: Page[];
}

export interface Page {
  title: string;
  id: string;
  part: string;
  sections: Section[];
}

export interface BaseSection {
  id: string;
  heading?: string;
  page: string;
}

export interface TextSection extends BaseSection {
  type: "text";
  content: string;
}

export interface ImageSection extends BaseSection {
  type: "img";
  content: string;
  uri: string;
}

export interface VideoSection extends BaseSection {
  type: "video";
  uri: string;
}

export interface FormSection extends BaseSection {
  type: "form";
  form: Input[];
  answer_lists?: AnswerList[];
}

export interface QuizSection extends BaseSection {
  type: "quiz";
  questions: Question[];
}

export type Nap = [Date, Date];

export interface DiaryEntry {
  id: string;
  date: Date;
  day_rating: number;
  naps: Nap[];
  sleep_aides: boolean;
  sleep_aides_detail: string;
  notes: string;
  sleep_quality: number;
  bedtime: Date;
  lights_out: Date;
  time_to_sleep: number;
  night_wakes: number[];
  waketime: Date;
  risetime: Date;
  finished?: boolean;
}
export interface SleepDiary {
  id: string;
  user: User;
  started_date: Date;
  diary_entries: DiaryEntry[];
}

export type Section =
  | TextSection
  | ImageSection
  | VideoSection
  | FormSection
  | QuizSection;
