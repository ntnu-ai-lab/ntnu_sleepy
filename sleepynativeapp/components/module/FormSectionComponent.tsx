import React from "react";
import { View } from "react-native";
import { FormSection } from "../../types/modules";

export function FormSectionComponent(props: {
    section: FormSection
}) {

    return (
        <View>
            
        </View>
    )  
}