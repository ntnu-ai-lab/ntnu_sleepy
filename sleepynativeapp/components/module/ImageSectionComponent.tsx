import React from "react";
import { View, StyleSheet, Image, Text } from "react-native";
import { colors } from "../../styles/styles";
import { ImageSection } from "../../types/modules";
import { Card } from "../material/Card";

export function ImageSectionComponent(props: { section: ImageSection }) {
  const { section } = props;

  const styles = StyleSheet.create({
    heading: {
      color: colors.text_white,
      fontSize: 20,
    },
    img: {
      width: 200,
      height: 200,
    },
    caption: {
      color: colors.text_secondary,
      fontSize: 12,
    },
  });

  return (
    <Card>
      {section.heading ? <Text style={styles.heading}>{section.heading}</Text> : <View />}
      <Image style={styles.img} source={{ uri: section.uri }} />
      <Text style={styles.caption}>{section.content}</Text>
    </Card>
  );
}
