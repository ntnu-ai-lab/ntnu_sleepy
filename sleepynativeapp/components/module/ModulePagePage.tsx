import React, { useEffect, useRef } from "react";
import { ScrollView, View, Text, TouchableOpacity } from "react-native";
import { Page, Section } from "../../types/modules";
import { FormSectionComponent } from "./section/FormSectionComponent";
import { ImageSectionComponent } from "./section/ImageSectionComponent";
import { TextSectionComponent } from "./section/TextSectionComponent";
import { VideoSectionComponent } from "./section/VideoSectionComponent";
import { PageTemplate } from "../material/PageTemplate";
import { IconButton } from "../material/IconButton";
import ArrowForward from "../../assets/arrowForward.svg";
import ArrowBack from "../../assets/arrowBack.svg";
import { useNavigation } from "@react-navigation/native";
import { moduleProgression } from "../../state/atoms";
import { useRecoilState } from "recoil";
import { QuizSectionComponent } from "./section/QuizSectionComponent";

export function ModulePagePage(props: {
  page: Page;
  navigatePage: (arg1: number) => void;
  navigatePart: (arg1: number) => void;
  pageProgression: number;
}) {
  const { page, navigatePage, pageProgression, navigatePart } = props;
  const scrollRef = useRef(null);
  const navigation = useNavigation();

  useEffect(() => {
    //@ts-ignore
    scrollRef.current?.scrollTo({ x: 0, y: 0, animated: true });
  }, [page]);

  return (
    <PageTemplate>
      <ScrollView ref={scrollRef}>
        {page ? (
          <View>
            {page.sections.map((s: Section, i) => {
              const type = s.type;
              switch (type) {
                case "img": {
                  return (
                    <View key={i}>
                      <ImageSectionComponent section={s} />
                    </View>
                  );
                }
                case "video": {
                  return (
                    <View key={i}>
                      <VideoSectionComponent section={s} />
                    </View>
                  );
                }
                case "form": {
                  return (
                    <View key={i}>
                      <FormSectionComponent section={s} />
                    </View>
                  );
                }
                case "text": {
                  return (
                    <View key={i}>
                      <TextSectionComponent section={s} />
                    </View>
                  );
                }
                case "quiz": {
                  return (
                    <View key={i}>
                      <QuizSectionComponent section={s} />
                    </View>
                  )
                }
              }
            })}
          </View>
        ) : (
          <View />
        )}

        <View
          style={{
            width: "100%",
            flexDirection: "row",
            justifyContent: "space-between",
            height: 50,
            paddingHorizontal: 20,
          }}
        >
          {pageProgression === 0 ? (
            <View />
          ) : (
            <TouchableOpacity onPress={() => navigatePage(-1)}>
              <View
                style={{ flexDirection: "row-reverse", alignItems: "center" }}
              >
                <Text style={{ fontSize: 18 }}>Forrige</Text>
                <ArrowBack />
              </View>
            </TouchableOpacity>
          )}
          {pageProgression === 1 ? (
            <TouchableOpacity
              onPress={() => { 
                navigatePart(1)
                //@ts-ignore because navigation wack as fuck
                navigation.goBack();
              }}
            >
              <View style={{ flexDirection: "row", alignItems: "center" }}>
                <Text style={{ fontSize: 18 }}>Fullf√∏r</Text>
                <ArrowForward />
              </View>
            </TouchableOpacity>
          ) : (
            <TouchableOpacity onPress={() => navigatePage(1)}>
              <View style={{ flexDirection: "row", alignItems: "center" }}>
                <Text style={{ fontSize: 18 }}>Neste</Text>
                <ArrowForward />
              </View>
            </TouchableOpacity>
          )}
        </View>
        <View style={{ height: 80 }} />
      </ScrollView>
    </PageTemplate>
  );
}
