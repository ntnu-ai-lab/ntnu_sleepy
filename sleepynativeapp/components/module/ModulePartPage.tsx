import React, { useState } from "react";
import { Text, View } from "react-native";
import { Part } from "../../types/modules";
import { PageTemplate } from "../material/PageTemplate";
import { ProgressBar } from "../material/ProgressBar";
import { ModulePagePage } from "./ModulePagePage";
import { IconButton } from "../material/IconButton";

export function ModulePartPage(props: { part: Part }) {
  const { part } = props;
  const [currentPage, setCurrentPage] = useState<number>(0);
  const numberOfPages = part.pages.length;

  function navigatePage(arg1: number) {
    console.log(currentPage, numberOfPages)
    if (currentPage === 0 && arg1 === -1) return
    if (currentPage === (numberOfPages - 1) && arg1 === 1) return 
    setCurrentPage(currentPage + arg1)
  }

  return (
    <PageTemplate>
      <View style={{ margin: 5 }}>
        <Text style={{ fontSize: 18, paddingBottom: 5 }}>
          Velkommen til denne delen!
        </Text>
        <ProgressBar percentage={((currentPage + 1) / numberOfPages) * 100} />
      </View>
      <View>
        <ModulePagePage page={part.pages[currentPage]} navigatePage={navigatePage}/>
      </View>
    </PageTemplate>
  );
}
