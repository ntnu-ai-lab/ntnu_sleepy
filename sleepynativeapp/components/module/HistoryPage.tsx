import { useNavigation, useRoute } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import React, { useEffect, useState } from "react";
import { View, Text, StyleSheet } from "react-native";
import { useRecoilValue } from "recoil";
import { cachedModules, moduleProgression } from "../../state/atoms";
import { colors } from "../../styles/styles";
import { ModuleExpanded } from "../../types/modules";
import { Button } from "../material/Button";
import { Card } from "../material/Card";
import { PageTemplate } from "../material/PageTemplate";
import { ModulePartPage } from "./ModulePartPage";

const styles = StyleSheet.create({
  heading: {
    fontSize: 20,
    color: colors.text_white,
    alignSelf: "center",
  },
  breadText: {
    color: `${colors.text_white}EF`
  }
});

export function HistoryPage() {
  const navigation = useNavigation();
  const finishedModules = useRecoilValue(cachedModules);

  return (
    <PageTemplate>
      {finishedModules?.length === 0 ? <View>
          <Card>
            <Text style={[styles.heading, {marginBottom: 15}]}>Ingen moduler å vise</Text>
            <Text style={styles.breadText}>Du har ikke sett på noen moduler enda, gå gjennom de i Hjem og så vil de dukke opp her.</Text>
          </Card>
        </View> : <View />}
      <View style={{ flexDirection: "column-reverse" }}>
        {finishedModules?.map((m) => {
          return (
            <Card key={m.id}>
              <Text style={styles.heading}>{m.title}</Text>
              <Button
                variant="contained"
                onClick={() => {
                  //@ts-ignore
                  navigation.navigate("home", {
                    screen: "History",
                    params: {
                      screen: "modulePartOverview",
                      params: { module: m },
                    },
                  });
                }}
              >
                <Text>Se modul</Text>
              </Button>
            </Card>
          );
        })}
      </View>
    </PageTemplate>
  );
}

export function HistoryModulePartOverview() {
  const route = useRoute(); //@ts-ignore
  const module: ModuleExpanded = route.params.module;
  const navigation = useNavigation();

  return (
    <PageTemplate>
      <View>
        {module?.parts.length === 0 ? <View>
          <Card>
            <Text style={[styles.heading, {marginBottom: 15}]}>Ingen deler å vise</Text>
            <Text style={styles.breadText}>Du har ikke sett på noen å delene i denne module, gå gjennom de i Hjem og så vil de dukke opp her.</Text>
          </Card>
        </View> : <View />}
        {module?.parts.map((p, i) => {
          return (
            <Card key={p.id + i}>
              <Text style={styles.heading}>{p.title}</Text>
              <Button
                variant="contained"
                onClick={() => {
                  //@ts-ignore
                  navigation.navigate("home", {
                    screen: "History",
                    params: {
                      screen: "modulePart",
                      params: { part: p, isHistory: true },
                    },
                  });
                }}
              >
                <Text>Se del</Text>
              </Button>
            </Card>
          );
        })}
      </View>
    </PageTemplate>
  );
}

export function HistoryNavigation() {
  const Stack = createNativeStackNavigator();

  return (
    <View style={{ height: "100%" }}>
      <Stack.Navigator
        screenOptions={{
          headerShown: false,
          headerTransparent: true,
          statusBarTranslucent: true,
          statusBarColor: "transparent",
          animation: "none",
        }}
        initialRouteName="moduleHistory"
      >
        <Stack.Screen name="moduleHistory" component={HistoryPage} />
        <Stack.Screen
          name="modulePartOverview"
          component={HistoryModulePartOverview}
        />
        <Stack.Screen name="modulePart" component={ModulePartPage} />
      </Stack.Navigator>
    </View>
  );
}
