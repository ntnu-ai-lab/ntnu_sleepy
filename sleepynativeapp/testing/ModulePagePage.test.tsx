// @ts-nocheck
import React from "react";
import renderer, { ReactTestRenderer } from "react-test-renderer";
import { RecoilRoot } from "recoil";
import { ModulePagePage } from "../components/module/ModulePagePage";
import { Page } from "../types/modules";

describe("Test that a module page get created correctly", () => {
  let page: ReactTestRenderer;

  const modulePage: Page = {
    id: "test-page",
    module: "nonexistant",
    sections: [
      {
        id: "testText",
        heading: "testText",
        content: "Lorem ipsum dolor sit amet",
        page: "test-page",
        type: "text",
      },
      {
        id: "testImg",
        heading: "testImg",
        content: "Sleeping man",
        uri: "https://post.medicalnewstoday.com/wp-content/uploads/sites/3/2020/02/325353_2200-1200x628.jpg",
        page: "test-page",
        type: "img",
      },
      {
        id: "test-section",
        heading: "Test heading",
        form: [
          {
            id: "test-input",
            type: "text",
            name: "test-name-input",
            label: "Test Label",
            helptext: "Test helptext",
            value: "Test value", //????
            section: "test-section",
            answers: [
              {
                id: "test-answer-list",
                input: "test-section",
                answerList: "test-answer-list",
                value: "Test answer value",
              },
            ],
          },
          {
            id: "test-input",
            type: "text",
            name: "test-name-input",
            label: "Test Label",
            helptext: "Test helptext",
            value: "Test value",
            section: "test-section",
            answers: [
              {
                id: "test-answer-list",
                input: "test-section",
                answerList: "test-answer-list",
                value: "Test answer value",
              },
            ],
          },
          {
            id: "test-input",
            type: "text",
            name: "test-name-input",
            label: "Test Label",
            helptext: "Test helptext",
            value: "Test value",
            section: "test-section",
            answers: [
              {
                id: "test-answer-list",
                input: "test-section",
                answerList: "test-answer-list",
                value: "Test answer value",
              },
            ],
          },
          {
            id: "test-Check-input",
            type: "checkbox",
            name: "test-check-input",
            label: "Er du Gravid?",
            helptext: "Test helptext",
            value: "Test value",
            section: "test-section",
            answers: [
              {
                id: "test-answer-list",
                input: "test-section",
                answerList: "test-answer-list",
                value: "Test answer value",
              },
            ],
          },
          {
            id: "test-Check-input",
            type: "checkbox",
            name: "test-check-input",
            label: "Er du i fast jobb?",
            helptext: "Test helptext",
            value: "Test value",
            section: "test-section",
            answers: [
              {
                id: "test-answer-list",
                input: "test-section",
                answerList: "test-answer-list",
                value: "Test answer value",
              },
            ],
          },
          {
            id: "test-Check-input",
            type: "select",
            name: "test-check-input",
            label: "Jobbform",
            helptext: "Test helptext",
            value: "Test value",
            section: "test-section",
            options: [
              {
                value: "Skift",
                lable: "Skift",
              },
              {
                value: "Deltid",
                lable: "Deltid",
              },
              {
                value: "8-4",
                lable: "8-4",
              },
            ],
            answers: [
              {
                id: "test-answer-list",
                input: "test-section",
                answerList: "test-answer-list",
                value: "Test answer value",
              },
            ],
          },
        ],
        page: "test-page",
        type: "form",
        answer_lists: [
          {
            id: "test-answer-list",
            section: "test-section",
            user: "",
            answers: [
              {
                id: "test-answer-list",
                input: "test-section",
                answerList: "test-answer-list",
                value: "Test answer value",
              },
            ],
          },
        ],
      },
    ],
  };

  beforeEach(() => {
    page = renderer.create(<RecoilRoot><ModulePagePage page={modulePage} /></RecoilRoot>);
  });

  it("renders", async () => {
    expect(page.toJSON()).toMatchSnapshot();
  })

  it("creates correct sections", async () => {
    const tree = page.toJSON();
    expect(tree.children).toBeTruthy();
    expect(tree.children[0].children[0].children.length).toBe(3);
  })
});
