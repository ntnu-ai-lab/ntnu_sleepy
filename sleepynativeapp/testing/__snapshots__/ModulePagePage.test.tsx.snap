// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that a module page get created correctly renders 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#A5C9CA",
        "height": "100%",
        "paddingTop": 0,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <RCTScrollView>
    <View>
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#395B64",
                "borderRadius": 20,
                "margin": 10,
                "padding": 12,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
              }
            }
          >
            testText
          </Text>
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 14,
              }
            }
          >
            Lorem ipsum dolor sit amet
          </Text>
        </View>
      </View>
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#395B64",
                "borderRadius": 20,
                "margin": 10,
                "padding": 12,
              },
              Object {
                "alignItems": "center",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
              }
            }
          >
            testImg
          </Text>
          <Image
            source={
              Object {
                "uri": "https://post.medicalnewstoday.com/wp-content/uploads/sites/3/2020/02/325353_2200-1200x628.jpg",
              }
            }
            style={
              Object {
                "aspectRatio": 1.7777777777777777,
                "borderRadius": 20,
                "width": "95%",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#9999A9",
                "fontSize": 12,
              }
            }
          >
            Sleeping man
          </Text>
        </View>
      </View>
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#395B64",
                "borderRadius": 20,
                "margin": 10,
                "padding": 12,
              },
              undefined,
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 20,
                },
                Object {
                  "alignSelf": "center",
                },
              ]
            }
          >
            Test heading
          </Text>
          <View>
            <View
              style={
                Object {
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  }
                }
              >
                Test Label
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#E7F6F2",
                      "borderColor": "",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "height": 112,
                      "justifyContent": "flex-start",
                      "marginVertical": 5,
                      "maxWidth": "100%",
                      "paddingHorizontal": 10,
                      "paddingVertical": 5,
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              >
                <TextInput
                  multiline={true}
                  onChangeText={[Function]}
                  placeholder="Test value"
                  placeholderTextColor="#9999A9"
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "height": "100%",
                    }
                  }
                  value=""
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#9999A9",
                    "fontSize": 12,
                  }
                }
              >
                Test helptext
              </Text>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  }
                }
              >
                Test Label
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#E7F6F2",
                      "borderColor": "",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "height": 112,
                      "justifyContent": "flex-start",
                      "marginVertical": 5,
                      "maxWidth": "100%",
                      "paddingHorizontal": 10,
                      "paddingVertical": 5,
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              >
                <TextInput
                  multiline={true}
                  onChangeText={[Function]}
                  placeholder="Test value"
                  placeholderTextColor="#9999A9"
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "height": "100%",
                    }
                  }
                  value=""
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#9999A9",
                    "fontSize": 12,
                  }
                }
              >
                Test helptext
              </Text>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  }
                }
              >
                Test Label
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "backgroundColor": "#E7F6F2",
                      "borderColor": "",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "height": 112,
                      "justifyContent": "flex-start",
                      "marginVertical": 5,
                      "maxWidth": "100%",
                      "paddingHorizontal": 10,
                      "paddingVertical": 5,
                      "width": "100%",
                    },
                    undefined,
                  ]
                }
              >
                <TextInput
                  multiline={true}
                  onChangeText={[Function]}
                  placeholder="Test value"
                  placeholderTextColor="#9999A9"
                  style={
                    Object {
                      "color": "#000000",
                      "fontSize": 16,
                      "fontWeight": "400",
                      "height": "100%",
                    }
                  }
                  value=""
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#9999A9",
                    "fontSize": 12,
                  }
                }
              >
                Test helptext
              </Text>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  }
                }
              >
                Er du Gravid?
              </Text>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#E7F6F2",
                    "borderRadius": 10,
                    "justifyContent": "center",
                    "marginRight": 50,
                    "width": 38,
                  }
                }
              >
                <View
                  accessibilityLiveRegion="polite"
                  accessibilityRole="checkbox"
                  accessibilityState={
                    Object {
                      "checked": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 18,
                        "padding": 6,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "opacity": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  }
                }
              >
                Er du i fast jobb?
              </Text>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#E7F6F2",
                    "borderRadius": 10,
                    "justifyContent": "center",
                    "marginRight": 50,
                    "width": 38,
                  }
                }
              >
                <View
                  accessibilityLiveRegion="polite"
                  accessibilityRole="checkbox"
                  accessibilityState={
                    Object {
                      "checked": false,
                      "disabled": false,
                    }
                  }
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "overflow": "hidden",
                      },
                      Object {
                        "borderRadius": 18,
                        "padding": 6,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "opacity": 0,
                      }
                    }
                  >
                    <Text />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "paddingVertical": 5,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  }
                }
              >
                Jobbform
              </Text>
              <View
                style={
                  Object {
                    "zIndex": 100,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    onLayout={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#E7F6F2",
                        "borderRadius": 20,
                        "display": "flex",
                        "flexDirection": "row",
                        "height": 56,
                        "justifyContent": "space-between",
                        "marginVertical": 5,
                        "maxWidth": "100%",
                        "paddingHorizontal": 15,
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontFamily": "System",
                            "fontWeight": "400",
                          },
                          Object {
                            "textAlign": "left",
                          },
                          Object {
                            "color": "#9999A9",
                            "fontSize": 16,
                            "fontWeight": "400",
                            "width": "80%",
                          },
                        ]
                      }
                    >
                      Jobbform
                    </Text>
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight={24}
                      bbWidth={24}
                      fill="none"
                      focusable={false}
                      height={24}
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "borderWidth": 0,
                          },
                          Object {
                            "flex": 0,
                            "height": 24,
                            "width": 24,
                          },
                        ]
                      }
                      vbHeight={24}
                      vbWidth={24}
                      width={24}
                      xml="<svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\">
  <path fill-rule=\\"evenodd\\" clip-rule=\\"evenodd\\" d=\\"M18.5303 9.53033L12.5303 15.5303C12.2374 15.8232 11.7626 15.8232 11.4697 15.5303L5.46968 9.53033L6.53033 8.46967L12 13.9393L17.4697 8.46967L18.5303 9.53033Z\\" fill=\\"black\\"/>
  </svg>
  "
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <RNSVGGroup
                        fill={null}
                        propList={
                          Array [
                            "fill",
                          ]
                        }
                      >
                        <RNSVGPath
                          clipRule={0}
                          d="M18.5303 9.53033L12.5303 15.5303C12.2374 15.8232 11.7626 15.8232 11.4697 15.5303L5.46968 9.53033L6.53033 8.46967L12 13.9393L17.4697 8.46967L18.5303 9.53033Z"
                          fill={4278190080}
                          fillRule={0}
                          propList={
                            Array [
                              "fill",
                              "fillRule",
                            ]
                          }
                        />
                      </RNSVGGroup>
                    </RNSVGSvgView>
                  </View>
                </View>
                <RCTScrollView
                  scrollToOverflowEnabled={true}
                  style={
                    Object {
                      "backgroundColor": "#E7F6F2",
                      "borderColor": "rgba(0, 0, 0, 0.24)",
                      "borderRadius": 10,
                      "borderWidth": 1,
                      "display": "none",
                      "maxHeight": 210,
                      "overflow": "hidden",
                      "paddingTop": 5,
                      "width": 0,
                      "zIndex": 15000,
                    }
                  }
                >
                  <View>
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "height": 40,
                          "justifyContent": "center",
                          "marginBottom": 1,
                          "opacity": 1,
                          "paddingHorizontal": 20,
                          "width": "100%",
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Object {
                              "fontSize": 16,
                            },
                          ]
                        }
                      >
                        Skift
                      </Text>
                    </View>
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "height": 40,
                          "justifyContent": "center",
                          "marginBottom": 1,
                          "opacity": 1,
                          "paddingHorizontal": 20,
                          "width": "100%",
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Object {
                              "fontSize": 16,
                            },
                          ]
                        }
                      >
                        Deltid
                      </Text>
                    </View>
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Object {
                          "alignItems": "flex-start",
                          "height": 40,
                          "justifyContent": "center",
                          "marginBottom": 1,
                          "opacity": 1,
                          "paddingHorizontal": 20,
                          "width": "100%",
                        }
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontWeight": "400",
                            },
                            Object {
                              "textAlign": "left",
                            },
                            Object {
                              "fontSize": 16,
                            },
                          ]
                        }
                      >
                        8-4
                      </Text>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
              <Text
                style={
                  Object {
                    "color": "#9999A9",
                    "fontSize": 12,
                  }
                }
              >
                Test helptext
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#A5C9CA",
                "borderColor": "",
                "borderRadius": 20,
                "borderWidth": 0,
                "justifyContent": "center",
                "marginVertical": 5,
                "minHeight": 56,
                "minWidth": 20,
                "opacity": 1,
                "overflow": "hidden",
              }
            }
          >
            <Text>
              Send in
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
